Added mcp-server name validation

preetDev004 opened the pull request:
This PR implements server name validation for MCP (Model Context Protocol) servers to ensure they conform to the required pattern ^[a-zA-Z0-9_-]+$. This addresses the TODO comment in mcp_connection_manager.rs:82.

+ Added validation before spawning MCP client tasks
+ Invalid server names are added to errors map with descriptive messages

I have read the CLA Document and I hereby sign the CLA

preetDev004 committed:
Added mcp-server name validation

preetDev004 commented:
@bolinfest added requested changes in this new PR

bolinfest commented on codex-rs/core/src/util.rs:
```diff
@@ -64,3 +64,9 @@ pub fn is_inside_git_repo(config: &Config) -> bool {
     false
 }
+
+pub fn is_valid_server_name(server_name: &str) -> bool {
```
I don't really like having generic "util" files because it's not clear what goes in there: please keep this in `mcp_connection_manager.rs` since that is the only place it is used.

preetDev004 replied:
@bolinfest I fixed it!

preetDev004 committed:
Removed clonning & moved the validation func from the util to mcp_connection_manager

bolinfest commented:
I think you need to run the formatter:

https://github.com/openai/codex/blob/5b820c5ce759158d274c22939eccc4839cb49c49/codex-rs/justfile#L25

preetDev004 committed:
formatted the file

preetDev004 commented:
> I think you need to run the formatter:
> 
> https://github.com/openai/codex/blob/5b820c5ce759158d274c22939eccc4839cb49c49/codex-rs/justfile#L25

I thought pre-commit took care of this, but never mind!

bolinfest committed:
Update mcp_connection_manager.rs

- move helper function to bottom of file
- rename function arg to be self-documenting

bolinfest committed:
Update mcp_connection_manager.rs

fix formatting (using the GH editor is hard...)

bolinfest committed:
Update mcp_connection_manager.rs

wow, I need a real editor
